### ISSUE TITLE
Segfault on ubuntu with mysql 5.7.16 and ruby 2.3.3

### ISSUE THREAD

**Issue by duduribeiro** [brianmario/mysql2#813]
Hi guys.

I'm currently having a problem with mysql2 (and I don't know if this problem is in mysql2 gem).
I'm using sidekiq 4.2.5 with mysql2 0.4.5 (I upgraded from 0.4.4 to test if this resolve the problem, but without success) and ruby 2.3.3 (upgraded from ruby 2.3.1 to test if this resolve the problem, but without success).

And my jobs works fine, until I receive this Segfault, and it kills my Sidekiq.
The stacktrace is here: http://pastebin.com/3pZUeeFG

I tried to found some issue with this behave but I haven't found anything.

Is this a problem with mysql2?

Thanks a lot 🍻

**Comment by sodabrew** [comment]
The relevant part of the stack trace, good news it had line numbers! I'll dig further in the new year:

```

-- C level backtrace information -------------------------------------------
/home/deploy/.rvm/rubies/ruby-2.3.3/bin/../lib/libruby.so.2.3(rb_vm_bugreport+0x4ea) [0x7f68eddc98da] vm_dump.c:692
/home/deploy/.rvm/rubies/ruby-2.3.3/bin/../lib/libruby.so.2.3(rb_bug_context+0xd4) [0x7f68edc5a2b4] error.c:435
/home/deploy/.rvm/rubies/ruby-2.3.3/bin/../lib/libruby.so.2.3(sigsegv+0x3e) [0x7f68edd37e0e] signal.c:890
/lib/x86_64-linux-gnu/libc.so.6 [0x7f68ed8484b0]
/home/deploy/threshold-dashboard/releases/97/vendor/bundle/ruby/2.3.0/gems/mysql2-0.4.5/lib/mysql2/mysql2.so(disconnect_and_raise+0x3d) [0x7f68e720423d] client.c:594
/home/deploy/.rvm/rubies/ruby-2.3.3/bin/../lib/libruby.so.2.3(rb_rescue2+0x1f8) [0x7f68edc631e8] eval.c:846
/home/deploy/threshold-dashboard/releases/97/vendor/bundle/ruby/2.3.0/gems/mysql2-0.4.5/lib/mysql2/mysql2.so(rb_query+0x14b) [0x7f68e72063eb] client.c:761
```

**Comment by sodabrew** [comment]
Resolved by #816

**Comment by duduribeiro** [comment]
Thanks a lot @sodabrew. I will test this commit in my project and check if it works well.

🍻 

Have a wonderful year 🎉
